### PR TITLE
feat: Add detection of missing strategies translations

### LINKED ISSUE #48: FRP: Add detection of missing strategies translations
- Missing translations errors should NOT be displayed in the error %
- Missing translations errors should ONLY be displayed once a toggle is enable
- Should use the yDaemon API to get the data

Initial issue here: https://github.com/yearn/yearn-vaults-descriptions/issues/42

### PR DESCRIPTION
**Pull request by karelianpie**
## Description

For each strategy of each vault check if there are any translations missing

## Related Issue

Closes https://github.com/yearn/ySync/issues/48

## Motivation and Context

We'd like to be warned  of missing strategies' translations

## How Has This Been Tested?

Ran locally and saw the missing translations being displayed, refer to screenshot below

## Screenshots (if appropriate):

ySync

### CODE REVIEW

File: contexts/useYearn.tsx [hunk 1]
```diff
@@ -50,20 +69,29 @@ export const YearnContextApp = ({children}: {children: ReactElement}): ReactElem
...
+		const	[fromAPI, _ledgerSupport, _riskFramework, strategies] = await Promise.all([
 			axios.get(`https://ydaemon.yearn.finance/${_chainID}/vaults/all?classification=any&strategiesRisk=withRisk`),
 			axios.get('https://raw.githubusercontent.com/LedgerHQ/app-plugin-yearn/develop/tests/yearn/b2c.json'),
-			axios.get('https://raw.githubusercontent.com/yearn/yearn-data-analytics/master/src/risk_framework/risks.json')
-		]) as [any, any, any];
+			axios.get('https://raw.githubusercontent.com/yearn/yearn-data-analytics/master/src/risk_framework/risks.json'),
+			axios.get(`https://ydaemon.yearn.finance/${_chainID}/meta/strategies?loc=all`)
+		]) as [any, any, any, any];
+
+		const LANGUAGES = [...new Set(Object.values(strategies.data).map(({localization}: any): string[] => localization ? Object.keys(localization) : []).flat())];
```
Comment (karelianpie) [inline contexts/useYearn.tsx:87]: Added this so we can dynamically add new languages and it will just work

File: contexts/useYearn.tsx [hunk 2]
```diff
@@ -50,20 +69,29 @@ export const YearnContextApp = ({children}: {children: ReactElement}): ReactElem
...
+
+		const STRATEGIES: {[key: string]: any} = {};
+
+		for (const strategyAddress of Object.keys(strategies.data)) {
+			STRATEGIES[toAddress(strategyAddress)] = strategies.data[strategyAddress];
+		}
```
Comment (karelianpie) [inline contexts/useYearn.tsx:93]: I had to map the given addresses because they were not checksummed

Comment (Majorfi) [reply contexts/useYearn.tsx:93]: I will check yDaemon to ensure addresses are checksummed

File: components/VaultBox.tsx [hunk 3]
```diff
@@ -261,6 +261,32 @@ function	VaultBox({vault, settings, noStrategies}: {vault: any, settings: TSetti
 				</section>
 			)}
 
+			{Object.keys(aggregatedData[toAddress(vault.address)]?.missingTranslations).length !== 0 && settings.shouldShowMissingTranslations ? (
```
Comment (Majorfi) [inline components/VaultBox.tsx]: Crash on Optimism here, this should fix it:

```suggestion
			{Object.keys((aggregatedData?.[toAddress(vault.address)]?.missingTranslations) || []).length !== 0 && settings.shouldShowMissingTranslations ? (
```

**Review by DarkGhost7 — DISMISSED** [review]
lgtm is working locally. Great job on this!

**Review by Majorfi — DISMISSED** [review]

**Review by Majorfi — APPROVED** [review]

**Pull request merged**